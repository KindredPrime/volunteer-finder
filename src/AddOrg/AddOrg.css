.AddOrg__description {
  display: flex;
  align-items: top;
  justify-content: center;
  margin: 10px 0px 10px 0px;
}

@media screen and (min-width: 1200px) {
  .AddOrg__description {
    margin: 15px 0px 15px 0px;
  }
}