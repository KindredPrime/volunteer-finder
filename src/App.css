@import url('https://fonts.googleapis.com/css2?family=Karla&family=Sura&family=Varela+Round&display=swap');

* {
  box-sizing: border-box;
}

body {
  background-color: #1A4161;
  color: white;
  margin-bottom: 20px;
}

h1 {
  text-shadow: -3px 0 black, 0 3px black, 3px 0 black, 0 -3px black;
  font-size: 34px;
  line-height: 48px;
}

h2 {
  text-shadow: -2px 0 black, 0 2px black, 2px 0 black, 0 -2px black;
  font-size: 26px;
  line-height: 36px;
}

p, legend {
  text-shadow: -1px 0 black, 0 1px black, 1px 0 black, 0 -1px black;
}

h1, h2, legend {
  font-family: 'Varela Round', sans-serif;
}

legend {
  font-size: 18px;
  line-height: 24px;
}

p, label {
  font-family: 'Karla', sans-serif;
}

p, label {
  font-size: 16px;
  line-height: 24px;
}

p {
  text-align: left;
}

section {
  border-top: 3px solid #747A82;
  margin: 20px 0px 20px 0px;
}

section > * {
  margin-left: 5%;
  margin-right: 5%;
}

button {
  background-color: #FF9100;
  color: black;
  cursor: pointer;
}

button:disabled {
  background-color: #747A82;
}

a {
  text-decoration: none;
  color: white;
}

a, button {
  font-family: 'Sura', serif;
}

form {
  padding: 10px;
}

fieldset, legend {
  margin-left: auto;
  margin-right: auto;
}

label {
  margin-right: 10px;
}

form button {
  margin-top: 10px;
}

fieldset {
  border: none;
}

legend {
  font-weight: bold;
}

p.error {
  color: #FF9100;
}

ul {
  padding: 0;
}

li {
  list-style: none;
}

.App {
  margin-top: 75px;
  text-align: center;
}

.App.expanded {
  margin-top: 138px;
}

@media screen and (min-width: 750px) {
  .App.expanded {
    margin-top: 75px;
  }
}

@media screen and (min-width: 1200px) {
  .App, .App.expanded {
    margin-top: 0px;
    margin-left: 17%;
  }
}