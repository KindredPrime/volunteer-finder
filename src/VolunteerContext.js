import React from 'react';

const VolunteerContext = React.createContext({
  orgs: [],
  causes: [],
  addOrg: () => {},
  deleteOrg: () => {},
  appError: null,
  // Is the app fetching data from the API?
  fetching: false
});

export default VolunteerContext;