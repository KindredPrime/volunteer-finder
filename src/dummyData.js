const dummyUsers = [
  {
    id: 1,
    email: 'email1@email.com',
    username: 'user1',
    password: 'password1',
    orgsAdded: [1],
    eventsAdded: [1, 2]
  },
  {
    id: 2,
    email: 'email2@email.com',
    username: 'user2',
    password: 'password2',
    orgsAdded: [2],
    eventsAdded: [3]
  },
  {
    id: 3,
    email: 'email3@email.com',
    username: 'user3',
    password: 'password3',
    orgsAdded: [3],
    eventsAdded: []
  },
  {
    id: 4,
    email: 'email4@email.com',
    username: 'user4',
    password: 'password4',
    orgsAdded: [],
    eventsAdded: []
  },
  {
    id: 5,
    email: 'email5@email.com',
    username: 'user5',
    password: 'password5',
    orgsAdded: [4],
    eventsAdded: [4]
  }
];

const dummyOrgs = [
  {
    id: 1,
    name: 'YMCA',
    website: 'https://www.ymca.com',
    phone: '1-800-555-7777',
    email: 'contact@ymca.com',
    address: '1 YMCA Street Real City, Real State',
    causes: [1, 2, 3],
    tags: [2, 3, 4, 6]
  },
  {
    id: 2,
    name: 'ASPCA',
    website: 'https://www.aspca.com',
    phone: '1-800-123-4567',
    email: 'contact@aspca.com',
    address: '1 ASPCA Street Real City, Real State',
    causes: [5, 2],
    tags: [2, 3, 4, 10]
  },
  {
    id: 3,
    name: 'National VOAD',
    website: 'https://www.nvoad.com',
    phone: '1-800-456-7890',
    email: 'contact@nvoad.com',
    address: '1 NVOAD Street Rivendell, Middle Earth',
    causes: [4, 2],
    tags: [2, 3, 4, 5, 7, 11]
  },
  {
    id: 4,
    name: 'National Air and Space Museum',
    website: 'https://www.airandspace.com',
    phone: '1-800-987-6543',
    email: 'contact@airandspace.com',
    address: '1 Space Street Air City, Outer Space',
    causes: [6, 7],
    tags: [1, 2, 3, 4, 6, 12]
  }
];

const dummyEvents = [
  {
    id: 1,
    name: 'Member Appreciation Day',
    organization: 1,
    location: 'Reston YMCA Center',
    date: 'February 20, 2020 00:00:00',
    duration: '4 hours',
    description: 'Greet YMCA members as they enter the facility and offer them snacks.  See https://www.website.com for more details.',
    causes: [2, 3],
    tags: [2, 4, 12]
  },
  {
    id: 2,
    name: 'Triathlon Assistance',
    organization: 1,
    location: 'Reston YMCA Center',
    date: 'January 27, 2020 00:00:00',
    duration: '4 hours',
    description: 'Help YMCA staff host their annual indoor triathlon: sign in triathlon participants, direct the participants to the different stages of the triathlon, and record times. See https://www.triathlon.com for more details.',
    causes: [2, 3],
    tags: [2, 3, 9]
  },
  {
    id: 3,
    name: 'Adoption Day',
    organization: 2,
    location: 'Times Square, New York City, New York',
    date: 'December 13, 2020 00:00:00',
    duration: '8 hours',
    description: 'Help the ASPCA run their annual adoption day for their local pet shelter.',
    causes: [5],
    tags: [2, 3, 9, 12]
  },
  {
    id: 4,
    name: 'Docent Training Class',
    organization: 4,
    location: '14390 Air and Space Museum Pkwy Chantilly, VA 20151',
    date: 'January 30, 2021 00:00:00',
    duration: '2 hours',
    description: 'This is a required class for anyone who wants to be a docent (tour guide) at the National Air and Space Museum.',
    causes: [6, 7],
    tags: [2, 3, 13]
  }
];

const dummyTags = [
  {
    id: 1,
    name: 'virtual'
  },
  {
    id: 2,
    name: 'in-person'
  },
  {
    id: 3,
    name: 'weekends'
  },
  {
    id: 4,
    name: 'weekdays'
  },
  {
    id: 5,
    name: 'nights'
  },
  {
    id: 6,
    name: 'group'
  },
  {
    id: 7,
    name: 'delivery'
  },
  {
    id: 8,
    name: 'cooking'
  },
  {
    id: 9,
    name: 'administration'
  },
  {
    id: 10,
    name: 'cleaning'
  },
  {
    id: 11,
    name: 'heavy lifting'
  },
  {
    id: 12,
    name: 'greeting'
  },
  {
    id: 13,
    name: 'training'
  }
];

const dummyCauses = [
  {
    id: 1,
    name: 'Youth'
  },
  {
    id: 2,
    name: 'Community'
  },
  {
    id: 3,
    name: 'Health'
  },
  {
    id: 4,
    name: 'Disaster Response'
  },
  {
    id: 5,
    name: 'Animals'
  },
  {
    id: 6,
    name: 'Museums'
  },
  {
    id: 7,
    name: 'Education'
  },
  {
    id: 8,
    name: 'Human Rights'
  },
  {
    id: 9,
    name: 'Arts'
  }
];

export {
  dummyUsers,
  dummyOrgs,
  dummyEvents,
  dummyTags,
  dummyCauses
};