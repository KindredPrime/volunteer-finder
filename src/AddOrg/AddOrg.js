import React, { Component } from 'react';
import PropTypes from 'prop-types';
import VolunteerContext from '../VolunteerContext';
import Nav from '../Nav/Nav';
import EntityCheckboxes from '../EntityCheckboxes/EntityCheckboxes';
import { setCheckboxValue } from '../util';
import ValidationError from '../ValidationError/ValidationError';
import './AddOrg.css';

class AddOrg extends Component {
  static contextType = VolunteerContext;

  // Only causes and tags that are checked are kept in the state
  state = {
    name: {
      touched: false,
      value: ''
    },
    website: {
      touched: false,
      value: ''
    }, 
    phone: {
      touched: false,
      value: ''
    },
    email: {
      touched: false,
      value: ''
    },
    address: {
      touched: false,
      value: ''
    },
    causes: {

    },
    tags: {

    },
    error: null
  };

  updateField = (fieldName, fieldValue) => {
    this.setState({
      [fieldName]: {
        touched: true,
        value: fieldValue
      }
    });
  };

  validateRequiredInput = (fieldName) => {
    const fieldValue = this.state[fieldName].value;

    if (fieldValue.trim().length === 0) {
      return `You must provide a ${fieldName}`;
    }
  }

  validateContactInfo = () => {
    const { website, phone, email, address } = this.state;

    if (
      website.value.trim().length === 0 
      && phone.value.trim().length === 0 
      && email.value.trim().length === 0
      && address.value.trim().length === 0) {
        return `You must fill out at least one contact info field`;
      }
  }

  orgExists = () => {
    const { name, website, phone, email, address } = this.state;
    const { orgs } = this.context;

    return orgs.find((org) => {
      const nameEx = new RegExp(`^${name.value.trim()}$`, 'i');
      if (nameEx.test(org.name)) {
        return true;
      }
      
      const websiteEx = new RegExp(`^${website.value.trim()}/*$`, 'i');
      if (websiteEx.test(org.website)) {
        return true;
      }

      const phoneEx = new RegExp(`^${phone.value.trim()}$`);
      if (phoneEx.test(org.phone)) {
        return true;
      }

      const emailEx = new RegExp(`^${email.value.trim()}$`, 'i');
      if (emailEx.test(org.email)) {
        return true;
      }

      const addressEx = new RegExp(`^${address.value.trim()}$`, 'i');
      if (addressEx.test(org.address)) {
        return true;
      }

      return false;
    });
  }

  handleSubmit = (e) => {
    e.preventDefault();

    const { name, website, phone, email, address, causes, tags } = this.state;
    const { user, addOrg } = this.context;

    if (this.orgExists()) {
      this.setState({
        error: new Error('The organization already exists')
      });
    }
    else {
      const newId = addOrg(
        name.value,
        website.value,
        phone.value,
        email.value,
        address.value,
        Object.entries(causes).map(([cause, __]) => cause), // Converts causes into an array
        Object.entries(tags).map(([tag, __]) => tag),
        user.username
      );

      this.props.history.push(`/org/${newId}`);
    }
  }

  render() {
    const { causes, tags } = this.context;
    const { name, website, phone, email, address, error } = this.state;

    return (
      <div className="AddOrg">
        <Nav />
        <header>
          <h1>Add an Organization</h1>
        </header>

        <form id="AddOrg__form" onSubmit={this.handleSubmit}>
          <div>
            <label htmlFor="org-name">Name*</label>
            <input 
              type="text"
              id="org-name"
              name="org-name"
              onChange={(e) => this.updateField('name', e.target.value)}
              required
            />
            {name.touched && <ValidationError message={this.validateRequiredInput('name')} />}
          </div>

          <br />

          <fieldset>
            <legend>Contact Info</legend>

            <div>
              <label htmlFor="website">Website</label>
              <input 
                type="text"
                id="website"
                name="website"
                onChange={(e) => this.updateField('website', e.target.value)}
              />
            </div>

            <br />

            <div>
              <label htmlFor="phone">Phone</label>
              <input 
                type="tel"
                id="phone"
                name="phone"
                onChange={(e) => this.updateField('phone', e.target.value)}
              />
            </div>

            <br />

            <div>
              <label htmlFor="email">Email</label>
              <input
                type="email"
                id="email"
                name="email"
                onChange={(e) => this.updateField('email', e.target.value)}
                onBlur={() => this.setState({
                  email: {
                    value: this.state.email.value,
                    touched: true
                  }
                })}
              />
            </div>

            <br />

            <div>
              <label htmlFor="address">Address</label>
              <input
                type="text"
                id="address"
                name="address"
                onChange={(e) => this.updateField('address', e.target.value)}
              />
            </div>

            {(website.touched && phone.touched && email.touched && address.touched) && 
              <ValidationError message={this.validateContactInfo()} />}
          </fieldset>

          {causes && causes.length > 0 && (
            <>
              <br />

              <EntityCheckboxes 
                entities={causes} 
                handleClick={setCheckboxValue('causes', this)} 
                type="causes"
                legend="Causes"
              />
            </>
          )}

          {tags && tags.length > 0 && (
            <>
              <br />

              <EntityCheckboxes 
                entities={tags} 
                handleClick={setCheckboxValue('tags', this)} 
                type="tags"
                legend="Tags"
              />
            </>
          )}

          <br />

          <button
            type="submit"
            disabled={
              this.validateRequiredInput('name')
              || this.validateContactInfo()}
          >
            Add Organization
          </button>
        </form>

        {error && <p className="AddOrg__error">{error.message}</p>}
      </div>
    );
  }
}

AddOrg.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func
  }).isRequired
};

export default AddOrg;