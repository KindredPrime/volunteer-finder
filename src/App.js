import React, { Component } from 'react';
import { Route, Redirect } from 'react-router-dom';
import { dummyUsers, dummyOrgs, dummyCauses, dummyTags } from './dummyData';
import VolunteerContext from './VolunteerContext';
import Home from './Home/Home';
import Login from './Login/Login';
import SignUp from './SignUp/SignUp';
import Account from './Account/Account';
import Organization from './Organization/Organization';
import PublicUser from './PublicUser/PublicUser';
import OrgSearch from './OrgSearch/OrgSearch';
import AddOrg from './AddOrg/AddOrg';
import './App.css';

class App extends Component {
  static contextType = VolunteerContext;

  state = {
    user: null,
    users: dummyUsers,
    orgs: dummyOrgs,
    causes: dummyCauses,
    tags: dummyTags
  };

  loginUser = (username, password) => {
    const { users } = this.state;

    const user = users.find((elem) => (
      elem.username === username && elem.password === password
    ));

    if (user) {
      window.localStorage.setItem('userId', user.id);
      this.setState({
        user
      });
    }
    else {
      return 'Invalid username and password combination';
    }
  }

  logoutUser = () => {
    window.localStorage.removeItem('userId');
    this.setState({
      user: null
    });
  }

  signUpUser = (email, username, password) => {
    const { users } = this.state;

    const id = users.length + 1;
    users.push({
      id,
      email,
      username,
      password,
      orgsAdded: []
    });

    this.setState({
      users
    });

    this.loginUser(username, password);

    return id;
  }

  addOrg = (name, website, phone, email, address, causes, tags) => {
    const { user, users, orgs } = this.state;
    const id = orgs.length + 1;
    const org = {
      id,
      name,
      website: website || 'Not Available',
      phone: phone || 'Not Available',
      email: email || 'Not Available',
      address: address || 'Not Available',
      causes,
      tags
    };

    const newUser = {
      ...user,
      orgsAdded: [...user.orgsAdded, org.name]
    };
    const newUsers = users.map((elem) => {
      if (elem.id === user.id) {
        return newUser
      }

      return elem;
    });
    this.setState({
      user: newUser,
      users: newUsers,
      orgs: [...orgs, org]
    });

    return id;
  }

  componentDidMount() {
    const storedId = window.localStorage.getItem('userId');
    if (storedId) {
      const { users } = this.state;
      const user = users.find((elem) => elem.id === parseInt(storedId));
      this.setState({
        user
      });
    }
  }

  render() {
    const { user, users, orgs, causes, tags } = this.state;
    const contextValue = {
      user,
      users,
      orgs,
      causes,
      tags,
      loginUser: this.loginUser,
      logoutUser: this.logoutUser,
      signUpUser: this.signUpUser,
      addOrg: this.addOrg
    };

    return (
      <VolunteerContext.Provider value={contextValue}>
        <div className="App">
          <Route exact path="/" component={Home} />
          <Route path="/login" render={({ history }) => (
            !user ? <Login history={history} /> : <Redirect to="/account" />
          )} />
          <Route path="/signup" component={SignUp} />
          <Route path="/account" render={() => user ? <Account /> : <Redirect to="/login" />} />
          <Route path="/org/:id" component={Organization} />
          <Route path="/user/:id" component={PublicUser} />
          <Route path="/org-search" render={() => <OrgSearch pageLimit={10} />} />
          <Route path="/add-org" render={({ history }) => (
            window.localStorage.getItem('userId') ? <AddOrg history={history}/> : <Redirect to="/login" />
          )} />
        </div>
      </VolunteerContext.Provider>
    );
  }
}

export default App;
