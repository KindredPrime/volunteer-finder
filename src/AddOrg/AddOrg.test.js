import ReactDOM from 'react-dom';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import AddOrg from './AddOrg';
import { dummyOrgs, dummyCauses } from '../dummyData';
import App from '../App';
import VolunteerContext from '../VolunteerContext';

describe.only('AddOrg Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <AddOrg history={{}} />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('goes to org page after successful add', async () => {
    render(
      <MemoryRouter initialEntries={['/add-org']}>
        <App />
      </MemoryRouter>
    );

    const causesLegend = 'Causes* (select at least one)';
    await waitFor(() => expect(screen.getByText(causesLegend)).toBeInTheDocument());

    const name = 'New Org';
    const website = 'http://www.new-org.com';
    const phone = '1-800-555-5555';
    const email = 'contact@new-org.com';
    const address = 'White House, D.C.';
    const description = 'New Org description';

    userEvent.type(screen.getByLabelText('Name*'), name);
    userEvent.type(screen.getByLabelText('Website'), website);
    userEvent.type(screen.getByLabelText('Phone'), phone);
    userEvent.type(screen.getByLabelText('Email'), email);
    userEvent.type(screen.getByLabelText('Address'), address);
    userEvent.type(screen.getByLabelText('Description*'), description);
    // Click the first checkbox in the Causes fieldset
    userEvent.click(screen.getByText(causesLegend).nextSibling.lastChild);
    userEvent.click(screen.getByRole('button', { name: 'Add Organization' }));

    await waitFor(() => expect(screen.getByText(name)).toBeInTheDocument());
    expect(screen.getByText(website)).toBeInTheDocument();
    expect(screen.getByText(`Phone: ${phone}`)).toBeInTheDocument();
    expect(screen.getByText(`Email: ${email}`)).toBeInTheDocument();
    expect(screen.getByText(`Address: ${address}`)).toBeInTheDocument();
    expect(screen.getByText(description)).toBeInTheDocument();

    userEvent.click(screen.getByRole('button', { name: 'Delete Organization' }));
    await waitFor(() => expect(screen.getByText('Search for Organizations')).toBeInTheDocument());
  });

  it('renders error when org already exists', () => {
    const contextValue = {
      orgs: dummyOrgs,
      causes: dummyCauses.slice(0, 1)
    };
    
    render(
      <BrowserRouter>
        <VolunteerContext.Provider value={contextValue}>
          <AddOrg history={{}} />
        </VolunteerContext.Provider>
      </BrowserRouter>
    );

    const org = dummyOrgs[0];
    userEvent.type(screen.getByLabelText('Name*'), org.org_name);
    userEvent.type(screen.getByLabelText('Website'), org.website);
    userEvent.type(screen.getByLabelText('Description*'), org.org_desc);
    userEvent.click(screen.getByLabelText(dummyCauses[0].cause_name));
    userEvent.click(screen.getByRole('button', { name: 'Add Organization' }));

    expect(screen.getByText('The organization already exists')).toBeInTheDocument();
  });

  it(`renders an error when 'Name' is missing`, () => {    
    render(
      <BrowserRouter>
        <AddOrg history={{}} />
      </BrowserRouter>
    );

    const nameInput = screen.getByLabelText('Name*');
    userEvent.type(nameInput, 'some text');
    nameInput.setSelectionRange(0, 9);
    userEvent.type(nameInput, '{backspace}');
    
    expect(screen.getByText('You must provide a name')).toBeInTheDocument();
  });

  it('renders an error when no contact info fields', () => {
    render(
      <BrowserRouter>
        <AddOrg history={{}} />
      </BrowserRouter>
    );

    const websiteInput = screen.getByLabelText('Website');
    userEvent.type(websiteInput, 'http://www.website.com');
    websiteInput.setSelectionRange(0, 22);
    userEvent.type(websiteInput, '{backspace}');

    const phoneInput = screen.getByLabelText('Phone');
    userEvent.type(phoneInput, '1-800-555-5555');
    phoneInput.setSelectionRange(0, 14);
    userEvent.type(phoneInput, '{backspace}');

    userEvent.click(screen.getByLabelText('Email'));

    const addressInput = screen.getByLabelText('Address');
    userEvent.type(addressInput, '1 Some Rd. Vienna, VA');
    addressInput.setSelectionRange(0, 21);
    userEvent.type(addressInput, '{backspace}');
    
    expect(screen.getByText('You must fill out at least one contact info field'))
      .toBeInTheDocument();
  });
});