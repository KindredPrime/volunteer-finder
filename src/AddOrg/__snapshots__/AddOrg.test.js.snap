// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddOrg Component goes to org page after successful add 1`] = `
<body>
  <div>
    <div
      class="App"
    >
      <div
        class="Organization"
      >
        <nav
          class="Nav"
        >
          <button
            class="Nav__expander"
            type="button"
          >
            Menu
          </button>
        </nav>
        <header>
          <h1>
            New Org
          </h1>
        </header>
        <section>
          <header>
            <h2>
              Contact Info
            </h2>
          </header>
          <p>
            Website: 
            <a
              href="http://www.new-org.com"
              rel="noreferrer"
              target="_blank"
            >
              http://www.new-org.com
            </a>
          </p>
          <p>
            Phone: 
            1-800-555-5555
          </p>
          <p>
            Email: 
            contact@new-org.com
          </p>
          <p>
            Address: 
            White House, D.C
          </p>
          <p />
        </section>
        <section>
          <header>
            <h2>
              Causes
            </h2>
          </header>
          <ul
            class="Organization__causes"
          >
            <li>
              Human Rights
            </li>
          </ul>
        </section>
        <section>
          <header>
            <h2>
              Tags
            </h2>
          </header>
          <p
            class="Organization__tags"
          >
            virtual
          </p>
        </section>
        <p>
          Created By: 
          <a
            href="/user/1"
          >
            user1
          </a>
        </p>
      </div>
    </div>
  </div>
</body>
`;
